components/suggest/src/db.rs

```diff
@@ -1213,6 +1218,22 @@ impl<'a> SuggestDao<'a> {
         Ok(())
     }
 
+    pub fn insert_dismissal(&self, url: &str) -> Result<()> {
+        self.conn.execute(
+            "INSERT OR IGNORE INTO dismissed_suggestions(url_hash)
```

Comment:
The raw AMP landing URLs might contain date time templates "%YYYYMMDDHH%", the Component populates those templates with proper values via `suggestion::cook_raw_suggestion_url()`. So, when Firefox records a dismissed URL, it should be the cooked URL rather than the raw one. We might need to process the URL before inserting it into the table otherwise it won't be sufficient to block future URLs generated from the same raw URL.

@0c0w3 Could you point us to code path where we handle this on Desktop? Maybe we can reuse it here.

Reply:
Ahh interesting, I definitely need to address that.  I would have thought we should be storing the raw URL though.  Won't the timestamp will change each time the suggestion is displayed?

Reply:
Did some digging, looks like we're hashing the original URLs (the pre-cooked one) when recording for dismissals on Desktop (sources: [here](https://searchfox.org/mozilla-central/source/browser/components/urlbar/UrlbarProviderQuickSuggest.sys.mjs#470-473) and [here](https://searchfox.org/mozilla-central/source/browser/components/urlbar/private/BlockedSuggestions.sys.mjs#38-60)), meaning that we could add a new field (e.g. `original_url: Option<String>`) to `Suggestion` in the Rust component, and populate it with the original URL if there is any template in it.

Reply:
Great!